src/main/keep-container/keep-content/keep-common-note/keep-common-note.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Observable, of, switchMap } from "rxjs";
+import { Note } from "../../../Data Types/Note";
+import { NoteService } from "../../../Service/note.service";
+import { MatDialog } from "@angular/material/dialog";
+import { KeepCommonEditorComponent } from "../keep-common-editor/keep-common-editor.component";
+
+@Component({
+  selector: 'app-keep-common-note',
+  templateUrl: './keep-common-note.component.html',
+  styleUrls: ['./keep-common-note.component.scss'],
+
+})
+export class KeepCommonNoteComponent implements OnInit {
+
+  notes$: Observable<Note[]>;
+  selectedNote: Note | null = null; // Initialize as null
+  @Input() isArchiveNotePresent: boolean = false;
+  @Input() isSearch: boolean = false;
+  searchValue: string = '';
+  dialogBoxOpen: boolean = false;
+
+  constructor(public dialog: MatDialog, private noteService: NoteService) {
+    this.notes$ = this.noteService.getNotes();
+  }
+
+  selectNoteForEditing(note: Note) {
+    this.selectedNote = note;
+    this.notes$ = of(this.noteService.changeHiddenStatus(note.id, true));
+    this.openDialog();
+  }
+
+  formattedContent(content: string): string {
+    let updateContent: string = this.convertNewlinesToBreaks(content);
+    return this.highlightSearchTerm(updateContent, this.searchValue);
+  }
+
+  highlightSearchTerm(text: string, searchTerm: string): string {
+    if (!searchTerm) {
+      return text;
+    }
+    const regex = new RegExp(searchTerm, 'gi');
+    return text.replace(regex, (match) => {
+      return `<span class="highlight">${match}</span>`;
+    });
+  }
+
+  handleMoreIconClick(event: MouseEvent, note: Note) {
+    note.showDropdown = !note.showDropdown;
+    event.stopPropagation();
+    note.isMoreIconClicked = !note.isMoreIconClicked;
+  }
+
+  ngOnInit(): void {
+    this.notes$ = this.noteService.getNotes(); // Subscribe to the observable
```

Comment:
Instead of setting the initial value of notes$ within the constructor and then immediately overwriting it in ngOnInit, you can initialize it directly in the constructor to improve code clarity.